Add CoronaSDK 2014.2511

thehunmonkgroup opened the pull request.

vitorgalvao commented:
[Are you sure this is `:freemium` and not `:gratis`](https://github.com/caskroom/homebrew-cask/blob/master/doc/CASK_LANGUAGE_REFERENCE.md#license-stanza-details)?

> a free app to access a paid service is still `:gratis`, not `:freemium`

thehunmonkgroup commented:
Here's the feature comparison between the paid and free versions: http://coronalabs.com/products/faqs/

Seems like freemium to me, and I'll defer to your judgement on what's the best label for this cask.

vitorgalvao commented on Casks/coronasdk.rb:
```diff
@@ -0,0 +1,11 @@
+  homepage 'http://coronalabs.com/products/corona-sdk'
+  license :freemium
+
+  app 'CoronaSDK'
```
This should be `suite`, not `app`, since you’re linking the directory.

vitorgalvao replied:
This should be `suite`, not `app`, since you’re linking the directory. Alternatively (which might be the best option), `app 'Corona Simulator.app'` should be used, in which case the cask should be renamed. Why are you linking the whole directory?

vitorgalvao commented:
I don’t use the SDK, so you’d be best to assess that. However, it seems to me that what you linked to is the business model, when what matter here is the license of the app itself. Are the restrictions in the app dictated by the service you pick, or does the app itself impose limitations?

thehunmonkgroup commented:
As far as I can tell, the restrictions are imposed by the service.

Also, I'm linking the entire directory because it includes their example code for using the SDK, and I'd prefer if that was also updated with the simulator.

So, given your feedback, would it work for submission if I changed :freemium -> :gratis and app -> suite?

If so I'll commit and push those changes.

vitorgalvao commented:
> As far as I can tell, the restrictions are imposed by the service.

Then it should be `:gratis`, yes.

> I'm linking the entire directory because it includes their example code for using the SDK, and I'd prefer if that was also updated with the simulator.

Keep in mind that when using `app` instead of `suite`, all those extra files will be untouched and will be kept up-to-date just the same. The rule to know which one to use is: “will you need to access those other files directly” (i.e. do you call them in any way other than through the app)? If the answer is *yes*, do `suite 'CoronaSDK'`; if the answer is *no*, do `app 'Corona Simulator.app'` (and rename the cask to `corona-simulator.rb`, with the first line being changed to `cask :v1 => 'corona-simulator' do`). I’ll leave this decision to you.

Please ping me when you’ve made the changes (don’t forget to [squash your commits](http://davidwalsh.name/squash-commits-git)), and I’ll merge this in.

Thank you.

thehunmonkgroup commented:
Ok, all set. I considered your feedback, and feel that suite is most appropriate for this package -- I do access the files outside the simulator app all the time.
